app/controllers/main_controller.rb

```diff
@@ -86,8 +86,7 @@ def login
     validation_result = validate_user(params[:user_login], params[:user_login], params[:user_password])
     unless validation_result[:error].nil?
       flash_message(:error, validation_result[:error])
-      redirect_to action: 'login'
-      return
+      render :login, locals: {user_login: params[:user_login]} and return
```

Comment:
In general, put spaces inside `{` and `}`, and don't use `and` (so keep the return on a separate line).